pkg/http/api.go

```diff
@@ -179,7 +179,7 @@ func (a *api) constructPubSubEndpoints() []Endpoint {
 	return []Endpoint{
 		{
 			Methods: []string{fasthttp.MethodPost, fasthttp.MethodPut},
-			Route:   "publish/{pubsubname}/{topic:*}",
+			Route:   "publish/{pubsubname}/{topic}",
```

Comment:
I recall that we allowed slash-included topic name so that we use `:*`.

@yaron2 @mchmarny / Don't we ?

Reply:
Well. This fixed the bug. Users should provide encoded / in http. It is one of those %something type of encoding.

Reply:
Yeah technically, url needs to be uuencoded chars. but, as I recall, we allows to use publish/{pubsubname}/topic/A/B. then we allowed to route to topic/A/B path of http server... 
if we change the behavior, it will be the breaking change :)

but I would like to double-check with @yaron2 @mchmarny

Reply:
You are correct, this needs to be reverted back. It breaks the custom route feature.

Reply:
Let me take another look at it. There is another way to fix it.

Reply:
From what I see, you already fixed it by validating if the params are empty and then returning an error.

To be clear, this happens if the user uses a trailing slash int their URL, like this `/publish/` in which case the `topic` param is empty. The check you added solves this, IMO.

Reply:
Correct, our code was already validating the presence of component, we were just missing the check on nil topic which can result from trailing slash on publish POST for example. this LGTM

Reply:
That check was only in the GRPC API. For HTTP, I tried to solve via the route setting.

Reply:
> Correct, our code was already validating the presence of component, we were just missing the check on nil topic which can result from trailing slash on publish POST for example. this LGTM

So `:*` is just for trailling slash? then this fix will be fine.

Reply:
No, * is not for trailing slash. The change needs to be reverted to include *.

Reply:
Put it back and added more unit tests for the HTTP route to make sure this bug does not come back when someone messes with the HTTP route again.

Reply:
+1 unittest